src/App.jsx

```diff
@@ -14,45 +16,84 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ABC = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGood(goods, sortField, isReversed) {
+  const preparedGoods = [...goods];
+
+  if (sortField === SORT_FIELD_ABC) {
+    preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
+
+  if (sortField === SORT_FIELD_LENGTH) {
+    preparedGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (isReversed === true) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortField, setSortField] = useState('');
+  const visibleGoods = getPreparedGood(goodsFromServer, sortField, isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_FIELD_ABC)}
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD_ABC,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setIsReversed(!isReversed)}
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': isReversed === false,
+          })}
+        >
+          Reverse
+        </button>
+
+        {(sortField !== '' || isReversed === true) && (
+          <button
+            onClick={() => {
+              setSortField('');
+              setIsReversed(false);
```

Comment:
it is bad practise to make anonymous arrow functions in jsx.
make handler in component and pass reference here